### PR TITLE
Callwithchat mobile: dismissed drawer after device selection

### PR DESCRIPTION
**Pull request by carocao-msft**
# What
Callwithchat mobile: make drawer dismissed after device selection

# Why
https://skype.visualstudio.com/SPOOL/_workitems/edit/2896714

# Test

https://user-images.githubusercontent.com/96077406/180506794-dc3039f5-ba78-44ec-8a08-b4016bdcce53.mov

https://user-images.githubusercontent.com/96077406/180510270-94f6d812-78d5-4773-a0ce-3409ed8e2db1.mov

### CODE REVIEW

**Comment by JamesBurnside** [comment]
Two things to consider:
* The mobile drawer is also used when a participant is clicked in callwithchat - can either add the same manual close logic to when they are clicked too, or consider instead changing the `_DrawerMenu` itself to trigger close when a menu item (that doesn't have a sub menu) is clicked
* We have some UI tests that may need updating if the drawer dismisses on click

**Comment by carocao-msft** [comment]
> Two things to consider:
> 
> * The mobile drawer is also used when a participant is clicked in callwithchat - can either add the same manual close logic to when they are clicked too, or consider instead changing the `_DrawerMenu` itself to trigger close when a menu item (that doesn't have a sub menu) is clicked
> * We have some UI tests that I think will need the test checks updated to reopen the drawer if the drawer dismisses on click

Updated PR with screen recording

**Review by JamesBurnside — APPROVED** [review]

File: packages/react-composites/src/composites/CallWithChatComposite/components/MoreDrawer.tsx [hunk 1]
```diff
@@ -107,8 +107,9 @@ export const MoreDrawer = (props: MoreDrawerProps): JSX.Element => {
         // But this is a known issue in our state.
         onSelectSpeaker(selected as AudioDeviceInfo);
       }
+      props.onLightDismiss();
     },
-    [speakers, onSelectSpeaker]
+    [speakers, onSelectSpeaker, props]
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/CallWithChatComposite/components/MoreDrawer.tsx:112]: avoid `props` as this means any prop change will cause this to be recreated.
Instead update the above `const { speakers, onSelectSpeaker } = props;`
to have the onLightDismiss:
`const { speakers, onSelectSpeaker, onLightDismiss } = props;`
and put `onLightDismiss` into the dependency array

File: packages/react-composites/src/composites/CallWithChatComposite/components/MoreDrawer.tsx [hunk 2]
```diff
@@ -142,8 +143,9 @@ export const MoreDrawer = (props: MoreDrawerProps): JSX.Element => {
         // But this is a known issue in our state.
         onSelectMicrophone(selected as AudioDeviceInfo);
       }
+      props.onLightDismiss();
     },
-    [microphones, onSelectMicrophone]
+    [microphones, onSelectMicrophone, props]
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/CallWithChatComposite/components/MoreDrawer.tsx:148]: (nit: same here on avoiding props in the dep array)

Comment (edwardlee-msft) [reply packages/react-composites/src/composites/CallWithChatComposite/components/MoreDrawer.tsx:148]: ^^ I second this as well

File: packages/react-composites/src/composites/CallWithChatComposite/components/MoreDrawer.tsx (hunk 1 above)
Comment (edwardlee-msft) [reply packages/react-composites/src/composites/CallWithChatComposite/components/MoreDrawer.tsx:112]: ^^ I second this as well!

**Review by edwardlee-msft — APPROVED** [review]

File: packages/react-composites/src/composites/CallWithChatComposite/components/MoreDrawer.tsx (hunk 2 above)
Comment (carocao-msft) [reply packages/react-composites/src/composites/CallWithChatComposite/components/MoreDrawer.tsx]: updated :)

File: packages/react-composites/src/composites/CallWithChatComposite/components/MoreDrawer.tsx (hunk 1 above)
Comment (carocao-msft) [reply packages/react-composites/src/composites/CallWithChatComposite/components/MoreDrawer.tsx]: updated :)

**Pull request merged**